packages/react-app/src/components/back-to-top.tsx

```diff
@@ -9,13 +10,30 @@ const TopRightCornerStyled = styled.div`
 `;
 
 export function BackToTop() {
+  const [scrolled, setScrolled] = useState(false);
+
+  useEffect(() => {
+    const scrollView = document.getElementById('scroll-view');
+    scrollView?.addEventListener('scroll', handleScroll);
+    return () => {
+      scrollView?.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  const handleScroll = (e: Event) => {
+    const scrollView = e.currentTarget as HTMLElement;
+    if (scrollView) setScrolled(scrollView.scrollTop > 0);
```

Comment:
i recommend use brackets `{` in `if's` that is good pratice to avoid mislead other people will maintence the code.

Plus it also help in review because it get less modified lines.